pkg/kubelet/cm/dra/plugin/plugin.go

```diff
@@ -110,15 +123,18 @@ func (h *RegistrationHandler) RegisterPlugin(pluginName string, endpoint string,
 		return err
 	}
 
-	// Storing endpoint of newly registered DRA Plugin into the map, where plugin name will be the key
-	// all other DRA components will be able to get the actual socket of DRA plugins by its name.
-	// By default we assume the supported plugin version is v1alpha3
-	draPlugins.add(pluginName, &plugin{
+	pluginInstance := &plugin{
 		conn:                    nil,
 		endpoint:                endpoint,
 		version:                 v1alpha3Version,
 		highestSupportedVersion: highestSupportedVersion,
-	})
+	}
+
+	// Storing endpoint of newly registered DRA Plugin into the map, where plugin name will be the key
+	// all other DRA components will be able to get the actual socket of DRA plugins by its name.
+	// By default we assume the supported plugin version is v1alpha3
+	draPlugins.add(pluginName, pluginInstance)
+	h.controller.addPlugin(pluginName, pluginInstance)
```

Comment:
I was thinking about it, but decided against it because long term I want to move managing the NodeResourceSlices out of kubelet and into each individual plugin, with access to the apiserver merely proxied by kubelet. The advantage of that will be that kubelet doesn't need to understand the API version used by the plugins and the control plane.

Therefore I don't want to make `draPlugin` (a mutex plus map) more complex because that complexity won't be needed for long.

Reply:
Ack